server/repository/src/migrations/v1_08_00/assets/asset.rs

```diff
@@ -0,0 +1,47 @@
+use crate::migrations::DATE;
+use crate::migrations::DATETIME;
+use crate::{migrations::sql, StorageConnection};
+
+pub(crate) fn migrate(connection: &StorageConnection) -> anyhow::Result<()> {
+    sql!(
+        connection,
+        r#"
+        CREATE TABLE asset (
+            id TEXT NOT NULL PRIMARY KEY,
+            store_id TEXT NOT NULL REFERENCES store (id), -- This serves as the location of the asset at least for now
+            serial_number TEXT NOT NULL, 
+            asset_category_id TEXT REFERENCES asset_category (id),
+            asset_type_id TEXT REFERENCES asset_type (id),
+            asset_catalogue_item_id TEXT REFERENCES asset_catalogue_item (id),
+            installation_date {DATE},
+            replacement_date {DATE},
+            deleted_datetime {DATETIME},
+            created_datetime {DATETIME} NOT NULL,
+            modified_datetime {DATETIME} NOT NULL,
+            UNIQUE (serial_number, deleted_datetime) --If something doesn't have a serial number, one can be generated?
```

Comment:
Worth verifying this comment. Feels like if we don't reliably have a serial number we'll need something to identify the asset, I guess people could just type `fridge1` as a serial number?

Reply:
I'm not sure that people will be populating serial number reliably; and while the number should exist and should be unique.. my experience is that this isn't very real world. I was thinking that it would be a nullable text field, which people may have filled in.

auto generating a serial would give incorrect data: we'd have the serial populated but it wouldn't be the serial of the actual asset, which could lead to confusion. better I think to have it blank.

Reply:
Agreed, I think we'll need something to identify which asset is which if there is no serial number.
I'll add a `name` field which can give a human readable way to identify it.

Reply:
Ideally we would be able to generate and assign unique asset numbers so the government can use Open mSupply as the source of truth for their asset register numbering. 

Typically I think countries would be printing out the unique asset numbers and sticking them onto the assets so that they can be tied back to the asset register. 

Serial numbers would typically be generated by the manufacturer of the asset and may or may not be present, as mentioned, so couldn't serve the same purpose as a centralised national asset number.

I'm unsure if `name` would serve the purpose either - I feel like users would populate that field with things like `Fridge in the green room` but that doesn't help if the fridge is later moved as there is no sticker on the asset saying `Fridge in the green room`. 

@richardmoizeau - thoughts?

Reply:
I think we'll be generating a unique number as part of the QR Code part, which I guess would be a stick over sorts. I guess we could probably make the number available somewhere obvious in the UI? Not currently in any mockups though.

Reply:
Ah, it's not super clear, but I think this is supposed to be the unique asset number: 

<img width="1118" alt="image" src="https://github.com/msupply-foundation/open-msupply/assets/19318937/67ac3786-fb1d-4624-b67c-b823429fa57f">

Reply:
Ok fair enough, have added a code field. Will probably use it for the QR code as well.